app/src/main/java/edu/wisc/ece/pinpoint/pages/newpin/NewPinFragment.java

```diff
@@ -0,0 +1,137 @@
+package edu.wisc.ece.pinpoint.pages.newpin;
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.viewpager2.widget.ViewPager2;
+
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ScrollView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.Task;
+import com.google.android.material.tabs.TabLayout;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.firestore.DocumentReference;
+
+import edu.wisc.ece.pinpoint.R;
+import edu.wisc.ece.pinpoint.data.Pin;
+import edu.wisc.ece.pinpoint.data.Pin.PinType;
+import edu.wisc.ece.pinpoint.pages.profile.ProfileFragmentAdapter;
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link NewPinFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ *
+ */
+public class NewPinFragment extends Fragment {
+
+
+    private TabLayout newpin_tabLayout;
+    private ViewPager2 newpin_viewPager;
+
+    private ScrollView newpin_scrollview;
+
+    private EditText newpin_inputeditlayout;
+
+    private Button dropButton;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_new_pin, container, false);
+    }
+
+
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        newpin_tabLayout = requireView().findViewById(R.id.newpin_tab_layout);
+        newpin_viewPager = requireView().findViewById(R.id.newpin_view_pager);
+        newpin_scrollview = requireView().findViewById(R.id.newpin_scrollview);
+        newpin_inputeditlayout = requireView().findViewById(R.id.newpin_inputeditlayout);
+        newpin_tabLayout.addTab(newpin_tabLayout.newTab().setText("Text"));
+        newpin_tabLayout.addTab(newpin_tabLayout.newTab().setText("Image"));
+        NewPinFragmentAdapter fragmentAdapter =
+                new NewPinFragmentAdapter(this.getChildFragmentManager(),
+                        newpin_tabLayout.getTabCount(), getLifecycle());
+        newpin_viewPager.setAdapter(fragmentAdapter);
+
+        dropButton = requireView().findViewById(R.id.drop_pin_button);
+        dropButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                createNewPin();
+            }
+        });
```

Comment:
We can replace this with a lambda expression to make it a lot cleaner.

Reply:
Good call, making that change.